[GvimExt] fix the memory leak

bryonglodencissp opened the pull request:
Passing one pointer into `realloc()` and assigning the result directly into that same pointer variable can cause a memory leak if the reallocation fails, because the original allocation will still exist. The correct way to do this is to use a temporary pointer variable.

Commit:
cmdStrW: nulled but not freed upon failure

[src/GvimExt/gvimext.cpp:1049]: (error) Common realloc mistake: 'cmdStrW' nulled but not freed upon failure

Found by https://github.com/bryongloden/cppcheck

mattn commented:
It is possibly double free.

https://github.com/vim/vim/pull/1074/commits/a27f9d445d201fe512eb75b1f523d43c6e3247a6#diff-37f9e8a402de9bbe16db9b79f1912e25R1097

Commit:
[GvimExt] update gvimext.cpp

Because there's a way to reach the second free(cmdStrW) from the first one, setting cmdStrW = NULL avoids the undefined behavior caused by double-free.

bryonglodencissp reviewed (commented):
LGTM

k-takata commented:
https://github.com/vim/vim/pull/1074/files#diff-37f9e8a402de9bbe16db9b79f1912e25L1051
NULL pointer access will occur at here.

mattn commented:
@bryongloden Not LGTM.

https://github.com/vim/vim/pull/1074/files#diff-37f9e8a402de9bbe16db9b79f1912e25R1061

Null reference. And I don't understand what you mean in this issue This is simple malloc/realloc and free combination to grow-up memory.

brammool commented:
No progress for several months, let's close this.
